### PR TITLE
Fix for STORE-660

### PR DESCRIPTION
**Pull request by Malintha**

### CODE REVIEW

File: apps/publisher/themes/default/js/create_asset.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ $(function () {
                 var $content = $(content).removeClass('loading-animation-big').addClass('loading-animation');
                 createButton.parent().append($content);
             });
-            populateTags(arr);
+            // populateTags(arr);
```
Comment (mushthaq33) [inline apps/publisher/themes/default/js/create_asset.js]: Remove commented code bocks

File: apps/publisher/themes/default/js/create_asset.js [hunk 2]
```diff
@@ -26,26 +26,26 @@ $(function () {
...
         //The Select2 plugin creates multiple entries in the FormData
         //object.These duplicates need to be removed and a single
         //_tags entry created in their place
+
         arr.forEach(function(entry){
             if(entry.name !== '_tags'){
                 modifiedArr.push(entry);
             } else {
                 tagEntry = entry;
+                if(tagEntry == undefined) {
```
Comment (mushthaq33) [inline apps/publisher/themes/default/js/create_asset.js]: Formatting issue

File: apps/publisher/themes/default/js/create_asset.js (hunk 1 above)
Comment (Malintha) [inline apps/publisher/themes/default/js/create_asset.js]: Fixed in https://github.com/wso2/carbon-store/pull/843

File: apps/publisher/themes/default/js/create_asset.js (hunk 2 above)
Comment (Malintha) [inline apps/publisher/themes/default/js/create_asset.js]: Fixed in https://github.com/wso2/carbon-store/pull/843

File: jaggery-modules/rxt/module/scripts/asset/asset.js [hunk 3]
```diff
@@ -577,7 +577,7 @@ var asset = {};
             if((query.hasOwnProperty(key)) && (key!='type')){
```
Comment (thushara35) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: The most expensing condition should come last.

File: jaggery-modules/rxt/module/scripts/asset/asset.js [hunk 4]
```diff
@@ -1160,9 +1160,9 @@ var asset = {};
         var asset = this.get(id);
         var tagged; //Assume that the tag will not be applied
         var utilsAPI = require('utils');
-        //If the user has provided a single tag then it should be 
+        //If the user has provided a single tag then it should be
         //assigned to an array to keep the registry invocation uniform
-        if (!utilsAPI.reflection.isArray(tags)) {
+        if (typeof tags === 'string') {
```
Comment (thushara35) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: Please justify...

Comment (Malintha) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: This either can be a single string or an array of strings.
Please refer : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof
http://stackoverflow.com/questions/27509/detecting-an-undefined-object-property

Comment (Malintha) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: This either can be a single string or an array of strings.
Please refer : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof
http://stackoverflow.com/questions/27509/detecting-an-undefined-object-property

File: jaggery-modules/rxt/module/scripts/asset/asset.js (hunk 3 above)
Comment (Malintha) [inline jaggery-modules/rxt/module/scripts/asset/asset.js]: There is no point of checking value of key if it does not exist in the query. AFAIK, If it does not exists, JS will short circuit and skip evaluating the value of key. 
https://en.wikipedia.org/wiki/Short-circuit_evaluation

**Pull request merged**